src/Agent.Listener/JobDispatcher.cs

```diff
@@ -614,6 +617,35 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, WorkerDisp
                                 var messageType = MessageType.CancelRequest;
                                 if (HostContext.AgentShutdownToken.IsCancellationRequested)
                                 {
+                                    var service = HostContext.GetService<IFeatureFlagProvider>();
+                                    var ffState = await service.GetFeatureFlagAsync(HostContext, "DistributedTask.Agent.FailJobWhenAgentDies", Trace);
+                                    if (ffState.EffectiveState == "On")
+                                    {
+                                        resultOnAbandonOrCancel = TaskResult.Failed;
+                                    }
+                                    try
+                                    {
+                                        var telemetryData = new Dictionary<string, string>
```

Comment:
Please extract PublishTelemetry, similar to other usages.  (can be local)

Reply:
Done